### PR TITLE
Bump rubocop ruby parser version to 2.3

### PR DESCRIPTION
**Pull request by adamruzicka**

### CODE REVIEW

File: test/support/middleware_example.rb [hunk 1]
```diff
@@ -99,7 +100,7 @@ def run
       end
 
       def filter_sensitive_data
-        output[:spell] = '***'
+        output[:spell] = '***'.dup
```
Comment (ezr-ondrej) [inline test/support/middleware_example.rb]: Should I want to know why? xD

**Comment by ezr-ondrej** [comment]
The dups are going to be necessary all around in Ruby 3.0? That's bit sad :thinking:

File: test/support/middleware_example.rb (hunk 1 above)
Comment (adamruzicka) [reply test/support/middleware_example.rb]: We do some modifications of these strings using `<<`. By adding the magic comment (`# frozen_string_literal: true`) all string literals become frozen by default and exceptions are raised when we try to modify them. `dup`ing a frozen string creates an un-frozen copy of it, which can be modified.

**Comment by ezr-ondrej** [comment]
And are we sure we `dup`ed enough? :rofl:

**Comment by ezr-ondrej** [comment]
I mean are we confident the test discovered all the frozen strings, that shouldn't have been frozen?

**Comment by adamruzicka** [comment]
Pretty sure, as far as I'm aware we don't use, let alone modify strings much.

**Pull request merged**